software/base/src/main/java/org/apache/brooklyn/entity/machine/MachineInitTasks.java

```diff
@@ -98,6 +101,7 @@ protected void openIptablesImpl(Iterable<Integer> inboundPorts, SshMachineLocati
             log.info("Opening ports in iptables for {} at {}", entity(), machine);
 
             List<String> iptablesRules = Lists.newArrayList();
+            String iptablesInstallCommands = "";
```

Comment:
No point in creating a task and ssh-ing if the command is empty, better set it to null and don't do an ssh call in this case at all.